Update detail information - 00_SIGNALduino.pm

HomeAutoUser opened the pull request:
* **Please check if the PR fulfills these requirements**
<!-- Please do not remove these checkboxes. Check them if you have done the action behind this point -->
- [x] Tests for the changes have been added / modified (needed for for bug fixes / features)
- [x] commandref has been added / updated (needed for bug fixes / features)
- [x] CHANGED has been updated (needed for bug fixes / features)

<!--Please specify if this is a bugfix, feature or update of docs and remove the other options -->

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
docs update (in the night for distraction)

* **What is the current behavior?** (You can also link to an open issue here)
only 
```
ccregAll: 

ccreg 00: 00 2E 2D 47 D3 91 3D 04 32 00 00 06 00 21 65 E8
ccreg 10: 57 C4 30 23 B9 00 07 00 18 14 6C 06 00 91 87 6B
ccreg 20: F8 B6 11 EF 2A 15 1F 41 00 59 7F 3F 88 31 0B
```

* **What is the new behavior (if this is a feature change)?**
- added detailed register information (based on the chip documentation)
- better understanding for the user and faster finding in the documentation

```
ccregAll: 

ccreg 00: 00 2E 2D 47 D3 91 3D 04 32 00 00 06 00 21 65 E8
ccreg 10: 57 C4 30 23 B9 00 07 00 18 14 6C 06 00 91 87 6B
ccreg 20: F8 B6 11 EF 2A 15 1F 41 00 59 7F 3F 88 31 0B

Configuration Register Detail (adr,desc,val):
0x00 IOCFG2__ - 0x00
0x01 IOCFG1__ - 0x2E
0x02 IOCFG0__ - 0x2D
0x03 FIFOTHR_ - 0x47
0x04 SYNC1___ - 0xD3
0x05 SYNC0___ - 0x91
0x06 PKTLEN__ - 0x3D
0x07 PKTCTRL1 - 0x04
0x08 PKTCTRL0 - 0x32
0x09 ADDR____ - 0x00
0x0A CHANNR__ - 0x00
0x0B FSCTRL1_ - 0x06
0x0C FSCTRL0_ - 0x00
0x0D FREQ2___ - 0x21
0x0E FREQ1___ - 0x65
0x0F FREQ0___ - 0xE8
0x10 MDMCFG4_ - 0x57
0x11 MDMCFG3_ - 0xC4
0x12 MDMCFG2_ - 0x30
0x13 MDMCFG1_ - 0x23
0x14 MDMCFG0_ - 0xB9
0x15 DEVIATN_ - 0x00
0x16 MCSM2___ - 0x07
0x17 MCSM1___ - 0x00
0x18 MCSM0___ - 0x18
0x19 FOCCFG__ - 0x14
0x1A BSCFG___ - 0x6C
0x1B AGCCTRL2 - 0x06
0x1C AGCCTRL1 - 0x00
0x1D AGCCTRL0 - 0x91
0x1E WOREVT1_ - 0x87
0x1F WOREVT0_ - 0x6B
0x20 WORCTRL_ - 0xF8
0x21 FREND1__ - 0xB6
0x22 FREND0__ - 0x11
0x23 FSCAL3__ - 0xEF
0x24 FSCAL2__ - 0x2A
0x25 FSCAL1__ - 0x15
0x26 FSCAL0__ - 0x1F
0x27 RCCTRL1_ - 0x41
0x28 RCCTRL0_ - 0x00
0x29 FSTEST__ - 0x59
0x2A PTEST___ - 0x7F
0x2B AGCTEST_ - 0x3F
0x2C TEST2___ - 0x88
0x2D TEST1___ - 0x31
0x2E TEST0___ - 0x0B
```

HomeAutoUser commented:
Guten Tag,
ich wollte bewusst nicht ` $rmsg ` modifizieren weil diese ja die Kurzforn beinhaltet.

```
ccreg 00: 0D 2E 2D 47 D3 91 3D 04 32 00 00 06 00 21 65 E8
ccreg 10: 57 C4 30 23 B9 00 07 00 18 14 6C 06 00 91 87 6B
ccreg 20: F8 B6 11 EF 2A 15 1F 41 00 59 7F 3F 88 31 0B
```

Die Kurzform ist sehr gut wenn man schnell das Register kompakt teilen möchte hier unter den Entwicklern beispielsweise. Da wäre die Langform (`$rmsg2`) sehr viele Informationen auf einmal.

Eine Variante, das ich sofort zur Verarbeitung die $rmsg sofort nutze um hinten dran die "aufbereitete Information" zu schreiben fand ich sehr umständlich weil da irgendwie die Kurzform immer leidete.
Schwierigkeit, der String enthält "Doppelleerzeichen" welche ich herausfiltere, split müsste zusätzlich auf 2 Bedingungen angesetzt werden wobei das array keine Leerelemente haben soll und meine Zielsetzung war es, die gleiche Anzahl Elemente im Array der Namen und der Inhalte zu besitzen ohne Leerzeichen bei den Values.

HomeAutoUser commented:
@sidey79, *smile*
ich musste mal noch einen drauflegen weil es scheinbar einige user gibt, die sich mit dem register beschäftigen wollen.

Habe fertig ;-)

elektron-bbs commented:
Naja, ich habe Einwände :-)
Genau genommen sind ja für die Modulationsart nur Bit 4-6 zuständig und Manchester wird mit Bit 3 aktiviert. Wenn, wie im gestrigen Beispiel Register 0x12 auf 0x16 gesetzt wird, zeigt es bei deiner Variante nichts an.
Kannst du die Unterstriche nicht durch Leerzeichen ersetzen?

HomeAutoUser commented:
@elektron-bbs 
> Kannst du die Unterstriche nicht durch Leerzeichen ersetzen?

Nein das geht nicht, weil dahinter JavaScript kommt und einige Regex, so das 2 Leerzeichen zu einem Zeilenumbruch werden. Alles schon versucht und deswegen erschienen mir die Unterstriche als akzeptabler Kompromiss.

> Genau genommen sind ja für die Modulationsart nur Bit 4-6 zuständig und Manchester wird mit Bit 3 aktiviert. Wenn, wie im gestrigen Beispiel Register 0x12 auf 0x16 gesetzt wird, zeigt es bei deiner Variante nichts an.

Die eingepflegten Werte sind mit dem TI Tool verifiziert und ebenso auch die Manchester Option.
Wenn Werte von wo anders mit einwirken und der Wert nicht zuordenbar ist, so steht er allein drin ohne Text.

elektron-bbs commented:
Bitte keine halben Sachen :-)
![MDMCFG2](https://user-images.githubusercontent.com/26026988/70643910-30c72c00-1c42-11ea-93fb-d0ad9ed44e54.png)

HomeAutoUser commented:
Halb ist es nicht. Es hilft nur kleinen „dummen Usern“ wie mich ;-)
DEM_DCFILT_OFF und SYNC_MODE Texte versuche ich zu verifizieren weil dies nur geht, wenn ich die Werte auch manuell ändern kann zum verifizieren. Sobald besondere Berechn7ngen einwirken hat solch ein Wert kein Sinn weil diese fehlen.

elektron-bbs commented:
Ja, aber wenn ich Register 0x12 auf 0x16 ist die Modulation GFSK, aber angezeigt wird keine.
Du brauchst doch nur Bit 7 auf 0 setzen, right shift 4 und kannst dann bequem die Modulation unabhängig von den anderen Bits ermitteln.
Manchester ist halt eingeschaltet, wenn Bit 3 auf 1 ist.

HomeAutoUser commented:
Ich habe nun nochmal tiefer geschaut und überlegt.

> An dieser Stelle verstehe ich nicht, warum die Informationen nicht direkt in rmsg gespeichert wird.
Worin besteht der Unterschied ob es ein paar Zeilen später an rmsg angehängt wird.

Schau es dir nun an, ich glaube wir verstanden uns falsch.

>Der Mehrwert dieser Information ist mir nicht ersichtlich.
Möchtest Du die Doku von TI hier einbringen? Wer auch immer es liest, könnte verwirrt werden, dass der SIGNALduino mit dem Packet Händler arbeiten kann, was ja nicht so ist.

Nein das war nicht meine Absicht. Einzigst, wollte ich die Modulationsart wiedergeben. Ich denke aber, dies als einzige Ausgabe wäre wenn in der ccconf angebracht weil an der jetzigen Stelle ja ALLE Werte des Registers wären und dies bei der Detailansicht unter MDMCFG2 wiederzugeben ist nur die halbe Wahrheit. Im Register MDMCFG2 stecken ja mehr Informationen als nur allein die Modulation.

>Kannst du die Unterstriche nicht durch Leerzeichen ersetzen?

Das habe ich nun extra für dich erörtert ;-)
Kleiner "Punkt", nun werden bei verbose 5 $rmsg der Text mit \n dargestellt und nicht als Zeilenumbruch. Dies könnte man anpassen wenn man eine 2. Variable noch dafür bestimmen würde.

`2019.12.12 14:15:39 3: nano_868Mhz: Read, msg: regexp=^ccreg 00: cmd=ccregAll msg=ccreg 00: 00 2E 2D 47 D3 91 3D 04 32 00 00 06 00 21 65 E8\nccreg 10: 57 C4 30 23 B9 00 07 00 18 14 6C 07 00 91 87 6B\nccreg 20: F8 B6 11 EF 2A 15 1F 41 00 59 7F 3F 88 31 0B \n\nConfiguration Register Detail (adr,desc,val):\n0x00 IOCFG2   - 0x00\n0x01 IOCFG1   - 0x2E\n0x02 IOCFG0   - 0x2D\n0x03 FIFOTHR  - 0x47\n0x04 SYNC1    - 0xD3\n0x05 SYNC0    - 0x91\n0x06 PKTLEN   - 0x3D\n0x07 PKTCTRL1 - 0x04\n0x08 PKTCTRL0 - 0x32\n0x09 ADDR     - 0x00\n0x0A CHANNR   - 0x00\n0x0B FSCTRL1  - 0x06\n0x0C FSCTRL0  - 0x00\n0x0D FREQ2    - 0x21\n0x0E FREQ1    - 0x65\n0x0F FREQ0    - 0xE8\n0x10 MDMCFG4  - 0x57\n0x11 MDMCFG3  - 0xC4\n0x12 MDMCFG2  - 0x30\n0x13 MDMCFG1  - 0x23\n0x14 MDMCFG0  - 0xB9\n0x15 DEVIATN  - 0x00\n0x16 MCSM2    - 0x07\n0x17 MCSM1    - 0x00\n0x18 MCSM0    - 0x18\n0x19 FOCCFG   - 0x14\n0x1A BSCFG    - 0x6C\n0x1B AGCCTRL2 - 0x07\n0x1C AGCCTRL1 - 0x00\n0x1D AGCCTRL0 - 0x91\n0x1E WOREVT1  - 0x87\n0x1F WOREVT0  - 0x6B\n0x20 WORCTRL  - 0xF8\n0x21 FREND1   - 0xB6\n0x22 FREND0   - 0x11\n0x23 FSCAL3   - 0xEF\n0x24 FSCAL2   - 0x2A\n0x25 FSCAL1   - 0x15\n0x26 FSCAL0   - 0x1F\n0x27 RCCTRL1  - 0x41\n0x28 RCCTRL0  - 0x00\n0x29 FSTEST   - 0x59\n0x2A PTEST    - 0x7F\n0x2B AGCTEST  - 0x3F\n0x2C TEST2    - 0x88\n0x2D TEST1    - 0x31\n0x2E TEST0    - 0x0B\n`

![test](https://user-images.githubusercontent.com/25631927/70716226-3e38f080-1cec-11ea-9073-f6e2cc1163a5.jpg)

HomeAutoUser commented:
![Zwischenablage-1](https://user-images.githubusercontent.com/25631927/70724971-de4a4600-1cfb-11ea-83f1-4a87d7ae0c4e.jpg)

mehr war nicht meine Absicht ;-)
Es dient dazu, das der User beim verstellen des Register auch die Möglichkeit hat, zu schauen was eingestellt ist an Modulationsart.

elektron-bbs commented:
Mit den Leerzeichen sieht das gleich viel besser aus :-)
Eine kleine Änderung hätte ich noch gern: 
Configuration Register Detail (adr, desc, val)
Configuration Register Detail (address, name, value)
Es wird ja keine Beschreibung ausgegeben, sondern nur der Name des Registers.

HomeAutoUser commented:
Deinen Wunsch habe ich soeben geändert :-)

Da ich mit dem Register setzen, testen und rücksetzen speben einiges versuche, finde ich nun die Ausgabe sehr hilfreich. Mir fällt auf, es gibt Umstände, wo die Registerwerte manchmal "wirbel".
Das müsste man aber separat mal beleuchten.

elektron-bbs commented:
Danke.
Das ab und an die Register nocht korrekt ausgelesen werden, habe ich bei einem SIGNALduino auch schon bemerkt. Ich bin aber noch nicht dahinter gekommen, woran das liegt.
Es gibt aber auch einige Register, die der CC1101 wahrscheinlich selbst ändert.

sidey79 commented:
Die Screenshots sind super, jetzt habe ich eine viel bessere Vorstellung davon, wo welche Anpassung sichtbar wird :)

HomeAutoUser commented:
@sidey79,
ebenso auch hier, was hat coveralls zu beanstanden?

sidey79 commented:
Eine Zeile die bisher mit Tests abgedeckt war ist jetzt angeblich nicht mehr abgedeckt.

Das ist seltsam, dass es genau eine ist.

HomeAutoUser commented:
Kann das die https://coveralls.io/builds/27597947/source?filename=FHEM%2F00_SIGNALduino.pm#L1453 Zeile 1453 sein weil dies mit unicov oder so gekennzeichnet ist?

sidey79 commented:
Das könnte gut sein.

HomeAutoUser commented:
Es war ja nur eine Idee weil ich da in das Uhrwerk schaue und soviel verstehe :-D
Wenn ich die Meldungen vergleiche mit dem anderen PR, so glaube ich nicht, das es daran liegt. das coveralls möchte uns was anderes damit sagen

*Schulterzuck*

sidey79 commented:
Ich habe mir den Umstand angesehen.

Die Zeile 1453 ist die, welche angeblich nicht mehr berücksichtigt wird.

Das war davor Zeile 1421:
https://coveralls.io/builds/27513332/source?filename=FHEM/00_SIGNALduino.pm

Die war da aber auch vorher nicht in einem Test berücksichtigt.
Zusammengefasst. Weder Zeile 1453 noch 1421 waren durch einen Test abgedeckt.

Entweder eine Falschmeldung oder die Zeilennummer ist quatsch

HomeAutoUser commented:
@sidey79 
>Ist denn der PR jetzt fertig? Wenn ja, dann hätte ich die Frage nach Anpassung
CHANGED

Ja, ist fertig.

changed ist schon vollzogen: 
```
12.12.2019
 00_SIGNALduino.pm: added modulationsinfo to ccconf, added commandref cpmmand raw
```

>und können wir etwas mittels Test verifizieren?

nein, denn der Test ccconf pürft ja schon bis String Modulation: Was anderes zum testen fällt mir da nicht ein.

Meinerseits GO
